backend/src/ee/services/scim/scim-service.ts

```diff
@@ -594,10 +596,15 @@ export const scimServiceFactory = ({
         },
         tx
       );
+
       await orgMembershipDAL.updateById(
         membership.id,
         {
-          isActive: active
+          isActive: active,
+          ...(active && {
+            role,
+            roleId
+          })
```

Comment:
this would replace the org role assignment of a user even when it wasn't triggered by a reprovision operation because `replaceScimUser` is invoked by `PUT /Users/:orgMembership` which is what's used to update user properties